src/core/executor.rs

```diff
@@ -91,6 +93,29 @@ impl Executor {
             }
         }
     }
+
+    /// Spawn a task on the pool, returning a future completing on task success.
+    ///
+    /// If the task panic, returns `Err(())`.
+    #[cfg(feature = "quickwit")]
+    pub fn spawn_blocking<T: Send + 'static>(
+        &self,
+        task: impl FnOnce() -> T + Send + 'static,
+    ) -> impl std::future::Future<Output = Result<T, ()>> {
+        match self {
+            Executor::SingleThread => Either::Left(std::future::ready(Ok(task()))),
+            Executor::ThreadPool(pool) => {
+                let (sender, receiver) = oneshot::channel();
+                pool.spawn(|| {
+                    // we don't care if the receiver was dropped
+                    let _ = sender.send(task());
```

Comment:
```suggestion
                    if sender.is_closed() {
                        return;
                    }
                    let task_result = task();
                    let _ = sender.send(task());
```

Reply:
we'd also need a test for cancellability. You can borrow the one from tantivy.

Reply:
there doesn't appear to be a `is_closed()` on oneshot's sender. There is on tokio's, but i doubt we want to add that dependency

Reply:
This sounds like something that could easily be added upstream in the `oneshot` crate? This way everyone would benefit from the improved API and since this the early cancellation check is an optimization, it could land when upstream releases the API addition. (Or a Git dependency on a fork could be used until then.)

Reply:
(There is also `async-oneshot` which does support this, but I think contributing to an existing dependency is better than vetting a completely new one?)

Reply:
I agree that it could get upstreamed. I'm implementing a wrapper to have it quickly, but will also submit a patch to `oneshot`